src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -7486,6 +7509,10 @@ uint32 Unit::MeleeDamageBonusTaken(Unit* attacker, uint32 pdamage, WeaponAttackT
             AddPct(TakenTotalMod, (*i)->GetAmount());
     }
 
+    // Versatility
+    if (GetSpellModOwner())
+        TakenTotalMod -= CalculatePct(1.0, GetSpellModOwner()->GetRatingBonusValue(CR_VERSATILITY_DAMAGE_TAKEN) + GetSpellModOwner()->GetTotalAuraModifier(SPELL_AURA_MOD_VERSATILITY));
```

Comment:
What? versatility is additive?

Reply:
Its not even that.. Dmg taken is 0.5% per 1% versatility.

Correct me if i'm wrong.. but.. why does he use CalculatePct to 1% from ALL Versatility stat? Its not like you only profit from 1% of your whole versatility

Reply:
TakenTotalMod is a multiplier, not percentage so your entire comment is wrong.

Reply:
Ahhh thats what i needed to know, its handled by the gt files, thx for the info :)

Reply:
need to check how it works ^^
Just was copying from Ashamane to get some feedback on this :>

Reply:
Yes it is additive

Reply:
instead of GetSpellModOwner()->GetTotalAuraModifier(SPELL_AURA_MOD_VERSATILITY))
you can also just select PLAYER_VERSATILITY_BONUS.

Reply:
You are mistaking something here, assume player has 1% damage reduction from versatility and Astral Shift (-40%). He will not be taking 41% less damage but the final damage will be `TAKEN = ORIGINAL * 0.6 * 0.99`

Reply:
If you have 6217 Versatility (CR). At level 110 the gt value ist 475.
Means DMG reduction is 13,09%.

If you get additional a 3% buff (SPELL_AURA_VERSATILITY_MOD).
Your VERSATILITY_BONUS = 3.

Your TotalDMG Reduction is 16,09% from Versatility.

Do you think i need to put the calculation somewhere else?

Reply:
damn it, you did not understand what i was saying at all.
versatility is not additive with other damage reduction sources - which is what your current code is doing

Reply:
Ok gonna take a deeper look into it when home.

Reply:
Got it at the end.